tests/integration/components/marker-layer-test.js

```diff
@@ -110,6 +110,32 @@ test('marker updates dragging', function(assert) {
    assert.equal(marker._layer.dragging.enabled(), false, 'marker dragging disabled');
 });
 
+test('marker dragging bound when icon changes', function(assert) {
+
+  this.set('markerCenter', locations.nyc);
+  this.set('draggable', true);
+  this.set('currentIconUrl', 'custom-url.png');
+  this.set('currentSize', 12);
+
+  this.render(hbs`
+    {{#leaflet-map zoom=zoom center=center}}
+      {{marker-layer location=markerCenter draggable=draggable
+        icon=(icon
+          iconUrl=currentIconUrl
+          iconSize=(point currentSize currentSize))}}
+    {{/leaflet-map}}
+  `);
+
+    //pre-conditions
+   assert.ok(marker._layer.dragging.enabled(), 'marker dragging enabled');
+
+   this.set('draggable', false);
+   this.set('currentIconUrl', 'another-custom-url.png');
+
+   assert.equal(marker._layer.options.icon.options.iconUrl, 'another-custom-url.png');
+   assert.equal(marker._layer.dragging.enabled(), false, 'marker dragging disabled');
+});
```

Comment:
The tests I envisioned would be:

1. test 1 - negative draggability is preserved on icon changes
  1. Start with icon 1 and draggable false
  2. assert that draggability is off (pre-condition)
  3. change to icon 2
  4. assert that draggability is still off
1. test 2 - positive draggability is preserved on icon changes
  1. Start with icon 1 and draggable true
  2. assert that draggability is on (pre-condition)
  3. change to icon 2
  4. assert that draggability is still on

Makes sense?